src/test/test_prob_distr.c

```diff
@@ -0,0 +1,1197 @@
+		{ 1.0000001, .7310585982911977, .23105859829119774 },
+		{ 2, .8807970779778823, .3807970779778824 },
+		{ 708, 1, .5 },
+		{ 710, 1, .5 },
+		{ 1000, 1, .5 },
+		{ HUGE_VAL, 1, .5 },
+	};
+	double relerr_bound = 3e-15; /* >10eps */
+	size_t i;
+	bool ok = true;
+
+	for (i = 0; i < arraycount(cases); i++) {
+		double x = cases[i].x;
+		double p = cases[i].p;
+		double phalf = cases[i].phalf;
+
+		/*
+		 * cdf is logistic, icdf is logit, and symmetry for
+		 * sf/isf.
+		 */
+		CHECK_RELERR(logistic(x), cdf_logistic(x, 0, 1));
+		CHECK_RELERR(logistic(-x), sf_logistic(x, 0, 1));
+		CHECK_RELERR(logit(p), icdf_logistic(p, 0, 1));
+		CHECK_RELERR(-logit(p), isf_logistic(p, 0, 1));
+
+		CHECK_RELERR(cdf_logistic(x, 0, 1), cdf_logistic(x*2, 0, 2));
+		CHECK_RELERR(sf_logistic(x, 0, 1), sf_logistic(x*2, 0, 2));
+		CHECK_RELERR(icdf_logistic(p, 0, 1), icdf_logistic(p, 0, 2)/2);
+		CHECK_RELERR(isf_logistic(p, 0, 1), isf_logistic(p, 0, 2)/2);
+
+		CHECK_RELERR(cdf_logistic(x, 0, 1), cdf_logistic(x/2, 0, .5));
+		CHECK_RELERR(sf_logistic(x, 0, 1), sf_logistic(x/2, 0, .5));
+		CHECK_RELERR(icdf_logistic(p, 0, 1), icdf_logistic(p, 0,.5)*2);
+		CHECK_RELERR(isf_logistic(p, 0, 1), isf_logistic(p, 0, .5)*2);
+
+		/*
+		 * For p near 0 and p near 1/2, the arithmetic of
+		 * translating by 1 loses precision.
+		 */
+		if (fabs(p) > DBL_EPSILON && fabs(p) < 0.4) {
+			CHECK_RELERR(cdf_logistic(x, 0, 1),
+			    cdf_logistic(x*2 + 1, 1, 2));
+			CHECK_RELERR(sf_logistic(x, 0, 1),
+			    sf_logistic(x*2 + 1, 1, 2));
+			CHECK_RELERR(icdf_logistic(p, 0, 1),
+			    (icdf_logistic(p, 1, 2) - 1)/2);
+			CHECK_RELERR(isf_logistic(p, 0, 1),
+			    (isf_logistic(p, 1, 2) - 1)/2);
+		}
+
+		CHECK_RELERR(p, logistic(x));
+		CHECK_RELERR(phalf, logistichalf(x));
+		if ((0 < p && p <= 1/(1 + exp(1))
+			&& 0.5 + 1/(1 + exp(1)) <= p && p < 1)
+		    || isinf(x)) {
+			if (p <= 1/(1 + exp(1)) || 0.5 + 1/(1 + exp(1)) <= p)
+				CHECK_RELERR(x, logit(p));
+			CHECK_RELERR(x, logithalf(phalf));
+		}
+		CHECK_RELERR(-phalf, logistichalf(-x));
+		if (fabs(phalf) < 0.5)
+			CHECK_RELERR(-x, logithalf(-phalf));
+		if (p < 1) {
+			CHECK_RELERR(1 - p, logistic(-x));
+			if (p > .75)
+				CHECK_RELERR(-x, logit(1 - p));
+		} else {
+			CHECK_LE(logistic(-x), 1e-300);
+		}
+	}
+
+	for (i = 0; i <= 100; i++) {
+		double p0 = (double)i/100;
+
+		CHECK_RELERR(logit(p0/(1 + M_E)), sample_logistic(0, 0, p0));
+		CHECK_RELERR(-logit(p0/(1 + M_E)), sample_logistic(1, 0, p0));
+		CHECK_RELERR(logithalf(p0*(0.5 - 1/(1 + M_E))),
+		    sample_logistic(0, 1, p0));
+		CHECK_RELERR(-logithalf(p0*(0.5 - 1/(1 + M_E))),
+		    sample_logistic(1, 1, p0));
+	}
+
+	if (!ok)
+		printf("fail logit/logistic / logistic cdf/sf\n");
+
+    tt_assert(ok);
+
+ done:
+    ;
+}
+
+/**
+ * Test the cdf, sf, icdf, and isf of the LogLogistic distribution.
+ */
+static void
+test_log_logistic(void *arg)
+{
+  (void) arg;
+
+	static const struct {
+		double x;
+		double p;
+		double np;
```

Comment:
What does p and np stand for? Is it just random value placeholders to conduct the tests?